tests/test_meshtastic_utils_async_helpers.py

```
@@ -192,3 +201,45 @@ def _bad_lookup(_sender):
+        timeout_seconds=0.1,
+        poll_seconds=0.01,
+    )
+
+    assert result == "ok"
+
+
+def test_wait_for_future_result_with_shutdown_aborts_when_shutting_down():
+    future = MagicMock()
+
+    with patch.object(mu, "shutting_down", True):
+        with pytest.raises(TimeoutError, match="Shutdown in progress"):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)